x-pack/plugins/observability_solution/inventory/public/pages/inventory_page/index.tsx

```diff
@@ -4,105 +4,74 @@
-            entityTypes: entityTypes?.length ? JSON.stringify(entityTypes) : undefined,
-            kuery,
-          },
-        },
-        signal,
-      });
-    },
-    [entityTypes, inventoryAPIClient, kuery, sortDirection, sortField]
+  const { view } = query;
+  const [inventoryState, setInventoryState] = useState<InventoryState | undefined>(() =>
+    view ? (decode(view) as InventoryState) : undefined
   );
 
-  useEffectOnce(() => {
-    const searchBarContentSubscription = searchBarContentSubject$.subscribe(
-      ({ refresh: isRefresh, ...queryParams }) => {
-        if (isRefresh) {
-          refresh();
-        } else {
-          inventoryRoute.push('/', {
-            path: {},
-            query: { ...query, ...queryParams },
-          });
-        }
-      }
-    );
-    return () => {
-      searchBarContentSubscription.unsubscribe();
-    };
-  });
+  useEffect(() => {
+    if (inventoryState) {
+      const encodedView = encode(inventoryState);
 
-  function handlePageChange(nextPage: number) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: { ...query, pageIndex: nextPage },
-    });
-  }
+      if (encodedView !== view) {
+        inventoryRoute.replace('/', {
+          path: {},
+          query: {
+            ...query,
+            view: encodedView,
+          },
+        });
+      }
+    }
+    // Only run the effect on a change for `inventoryState`. Other deps cause this to refire too many times.
+    // Probably a better way of doing this.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [inventoryState]);
 
-  function handleSortChange(sorting: EuiDataGridSorting['columns'][0]) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        sortField: sorting.id as EntityColumnIds,
-        sortDirection: sorting.direction,
-      },
-    });
-  }
+  const setGrouping = useCallback(
+    (field: string) => setInventoryState((state) => ({ ...state, groupBy: field })),
+    []
+  );
+  const setPagination = useCallback(
+    (group: string, nextPage: number) =>
+      setInventoryState((state) => ({
+        ...state,
+        pagination: { ...state?.pagination, [group]: nextPage },
+      })),
+    []
+  );
 
-  function handleTypeFilter(entityType: EntityType) {
-    inventoryRoute.push('/', {
-      path: {},
-      query: {
-        ...query,
-        // Override the current entity types
-        entityTypes: [entityType],
-      },
-    });
-  }
+  // If state or grouping itself is `undefined`, fallback to the default view
+  const currentGrouping = inventoryState?.groupBy || DEFAULT_GROUPING;
 
   return (
-    <EntitiesGrid
-      entities={value.entities}
-      loading={loading}
-      sortDirection={sortDirection}
-      sortField={sortField}
-      onChangePage={handlePageChange}
-      onChangeSort={handleSortChange}
-      pageIndex={pageIndex}
-      onFilterByType={handleTypeFilter}
-    />
+    <InventoryPageViewContextProvider
```

Comment:
I think we are giving too much responsibility to this component. IMHO this component must only look at the URL and render the Unifed or the Grouped version. Anything beyond that, should be handled by each component.

Reply:
I've split all this out into a hook, so no more context here.